fission-web-server/library/Fission/Web/Server/Types.hs

```diff
@@ -455,39 +571,20 @@ instance ServerDID Server where
 
 instance Server.DID.Publicize Server where
   publicize = do
-    AWS.MockRoute53 mockRoute53 <- asks awsMockRoute53
-
     Host host <- Reflective.getHost
     did       <- getServerDID
     zoneID    <- asks serverZoneID
-
     let
       ourURL         = URL (URL.DomainName . Text.pack $ baseUrlHost host) Nothing
       txtRecordURL   = URL.prefix' (URL.Subdomain "_did") ourURL
 
       txtRecordValue = textDisplay $ DID.Oldstyle did
 
-    if mockRoute53
-      then do
-        logInfo $ mconcat
-          [ "MOCK: Setting DNS setting "
-          , textDisplay txtRecordURL
-          , " to "
-          , txtRecordValue
-          ]
-
-        return ok
-
-      else
-        Route53.set Txt txtRecordURL zoneID (pure txtRecordValue) 10 <&> \case
-          Left err ->
-            Left err
-
-          Right resp -> do
-            let status' = view crrsrsResponseStatus resp
-            if status' < 300
-              then ok
-              else Left $ Web.Error.toServerError status'
+    PowerDNS.set TXT txtRecordURL (textDisplay zoneID) (pure txtRecordValue) 10 <&> \case
+      Left err ->
+        Left err
+      Right _ -> do
+        ok
```

Comment:
It's _fine,_ but also you don't need the `do` here.

I'd also recommend against using `<&> \case` over `>>= \case` in monadic contexts. GHC will optimize this in either case, but right now you can't just drop in and do this:

```haskell
PowerDNS.set TXT txtRecordURL (textDisplay zoneID) (pure txtRecordValue) 10 <&> \case
  Left err -> do
    logError "BOOM
    return $ Left err

  Right -> do
    logInfo "Success"
    return ok
```

...because `<&>` is an inverse `fmap` with arguments reversed. To do more than one "action" here, you'd need to switch to `>>=`.

I used to be a heavy user of `<&> \case` because it's shorter, but it's bitten _me_ a bunch ver the years, yelling at the type checker not realizing that I'm literally 3 characters away from having it compile.